app/models/piece.rb

```diff
@@ -42,8 +42,17 @@ def blocked?(x, y)
     end
 
     piece_path.each do |square|
-      return true if game.is_occupied?(square[0], square[1])
+      return true if game.square_occupied?(square[0], square[1])
     end
     false
   end
+
+  def move_to!(x, y)
+    return false unless valid_move?(x, y)
+    if square_occupied?(x, y)
+      target_piece = @game.piece.find_by(x_position: x, y_position: y)
```

Comment:
Oh, one more thing - this should be just `game` rather than `@game` (since it's calling the `game` method defined by `belongs_to :game` rather than accessing an instance variable `@game` that doesn't exist).